src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -230,125 +285,150 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count, Tas
 
 
 /*
- * ExecuteDistributedModify is the main entry point for modifying distributed
- * tables. A distributed modification is successful if any placement of the
- * distributed table is successful. ExecuteDistributedModify returns the number
- * of modified rows in that case and errors in all others. This function will
- * also generate warnings for individual placement failures.
+ * ExecuteTaskAndStoreResults executes the task on the remote node, retrieves
```

Comment:
ExecuteTaskAndStoreResults: It looks like the name and comment are obsolete. This function may not store the results anymore in case of `ConsumeQueryResults`. So, maybe change the name to `ExecuteRouterTask` or so?